plugins/optimization-detective/class-od-preload-link-collection.php

```diff
@@ -136,6 +136,27 @@ static function ( array $carry, array $link ): array {
 		);
 	}
 
+	/**
+	 * Adds media features to the links.
+	 *
+	 * @phpstan-param Link $link
+	 * @param array $link Link array.
+	 * @phpstan-return Link
+	 * @return array Link array with media features added.
+	 */
+	private function add_media_features( array $link ): array {
+		$media_features = array( 'screen' );
+		if ( null !== $link['minimum_viewport_width'] && $link['minimum_viewport_width'] > 0 ) {
+			$media_features[] = sprintf( '(min-width: %dpx)', $link['minimum_viewport_width'] );
+		}
+		if ( null !== $link['maximum_viewport_width'] && PHP_INT_MAX !== $link['maximum_viewport_width'] ) {
+			$media_features[] = sprintf( '(max-width: %dpx)', $link['maximum_viewport_width'] );
+		}
+		$link['attributes']['media'] = implode( ' and ', $media_features );
+
+		return (array) $link;
+	}
```

Comment:
However, instead of amending the link, how about just returning the media features string instead?

```suggestion
	private function get_media_attribute( array $link ): string {
		$media_features = array( 'screen' );
		if ( null !== $link['minimum_viewport_width'] && $link['minimum_viewport_width'] > 0 ) {
			$media_features[] = sprintf( '(min-width: %dpx)', $link['minimum_viewport_width'] );
		}
		if ( null !== $link['maximum_viewport_width'] && PHP_INT_MAX !== $link['maximum_viewport_width'] ) {
			$media_features[] = sprintf( '(max-width: %dpx)', $link['maximum_viewport_width'] );
		}
		return implode( ' and ', $media_features );
	}
```

Reply:
Further still, what if this logic were rather incorporated into `get_adjacent_deduplicated_links()`? Instead of returning the result of the `array_reduce()` it could then loop over the links and construct the `media` attribute for each. This method could then be changed from `get_adjacent_deduplicated_links()` to something like `prepare_links()`.

Reply:
This approach seems better. Will update the `get_adjacent_deduplicated_links ()` function.